akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/RouteDirectivesSpec.scala

```diff
@@ -157,6 +158,30 @@ class RouteDirectivesSpec extends AnyFreeSpec with GenericRoutingSpec {
     }
   }
 
+  "the fromFunction directive" should {
+    "use a function to complete a request" in {
+      Get(Uri("https://akka.io/foo")) ~> {
+        fromFunction(req => Future.successful(HttpResponse(OK, entity = req.uri.toString)))
```

Comment:
Is 'fromFunction' the right name? It seems rather 'technical', but looking for a more 'functional' name I can only come up with rather generic terms like 'handle'.

There is already `Route` apply method that takes a function, so we should probably not overload that ;).